include/engine/engine.hpp

```diff
@@ -125,6 +127,7 @@ template <typename Algorithm> class Engine final : public EngineInterface
     }
     std::unique_ptr<DataFacadeProvider<Algorithm>> facade_provider;
     mutable SearchEngineData<Algorithm> heaps;
+    tbb::task_scheduler_init task_scheduler;
```

Comment:
Needs to be constructed with num threads otherwise default ctor will use available cpus to infer number - which will be wrong e.g. in a Docker container: 

https://www.threadingbuildingblocks.org/docs/help/hh_goto.htm?index.htm#reference/task_scheduler/task_scheduler_init_cls.html

Reply:
@daniel-j-h i added `use_threads_number` parameter that can be set in node bindings and in command line arguments of `osrm-routed`. `osrm-routed` now has two separate thread pools: server and internal tbb, so number of threads is actually doubled.

Reply:
It it? I thought the benefit of using the dynamically linked libtbb was to avoid exactly these issues?